src/raft/fixture.c

```diff
@@ -247,13 +297,57 @@ static void ioFlushAppend(struct io *s, struct append *append)
 	s->entries = entries;
 	s->n += append->n;
 
+	sm_move(&append->req->sm, FIO_END);
+
 done:
+	sm_fini(&append->req->sm);
 	if (append->req->cb != NULL) {
 		append->req->cb(append->req, status);
 	}
 	raft_free(append);
 }
 
+static void ioFlushTruncate(struct io *io, struct truncate *r)
+{
+	size_t n = (size_t)(r->index - 1); /* Number of entries left after truncation */
+
+	if (n > 0) {
+		struct raft_entry *entries;
+
+		/* Create a new array of entries holding the non-truncated
+		 * entries */
+		entries = raft_malloc(n * sizeof *entries);
+		assert(entries != NULL);
+		memcpy(entries, io->entries, n * sizeof *io->entries);
+
+		/* Release any truncated entry */
+		if (io->entries != NULL) {
+			size_t i;
+			for (i = n; i < io->n; i++) {
+				raft_free(io->entries[i].buf.base);
+			}
+			raft_free(io->entries);
+		}
```

Comment:
Marked code and code starting from line 334 look like copy-paste. 
I see a potential to make the code around it better.

Reply:
Indeed this can be written much more compactly, thanks!